xdsl/transforms/convert_memref_to_ptr.py

```diff
@@ -0,0 +1,164 @@
+from collections.abc import Iterable
+from dataclasses import dataclass
+from typing import cast
+
+from xdsl.context import MLContext
+from xdsl.dialects import arith, builtin, memref, ptr
+from xdsl.ir import Operation, SSAValue
+from xdsl.irdl import Any
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    GreedyRewritePatternApplier,
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+from xdsl.utils.exceptions import DiagnosticException
+
+
+def offset_calculations(
+    memref_type: memref.MemRefType[Any], indices: Iterable[SSAValue]
+) -> tuple[list[Operation], SSAValue]:
+    """Get operations calculating an offset which needs to be added to memref's base pointer to access an element referenced by indices."""
+
+    assert isinstance(memref_type.element_type, builtin.FixedBitwidthType)
+
+    match memref_type.layout:
+        case builtin.NoneAttr():
+            strides = builtin.ShapedType.strides_for_shape(memref_type.get_shape())
+        case builtin.StridedLayoutAttr():
+            strides = memref_type.layout.get_strides()
+        case _:
+            raise DiagnosticException(f"Unsupported layout type {memref_type.layout}")
```

Comment:
Oh I forgot about this, could you please add a small PR with a test that catches this diagnostic exception?